core/src/avm2.rs

```diff
@@ -314,7 +314,13 @@ impl<'gc> Avm2<'gc> {
             .map(|e| e.event_type())
             .unwrap_or_else(|| panic!("cannot dispatch non-event object: {:?}", event));
 
-        let mut activation = Activation::from_nothing(context.reborrow());
+        let domain = context
+            .library
+            .library_for_movie(context.swf.clone())
```

Comment:
This is the top-level movie, which isn't necessarily correct here - there's nothing stopping us from calling this function with a `target` in a different domain. I need to write some tests to figure out what Flash Players's behavior is here.

Reply:
Should I use script domain instead then?